app/catalog/ProductCard.tsx

```diff
@@ -0,0 +1,108 @@
+// import React from 'react';
+// import Image from 'next/image';
+// import { Product, ProductData } from './constants';
+// import Link from 'next/link';
+
+// interface ProductCardProps {
+//   product: ProductData;
+//   index: number;
+// }
+// const ProductCard: React.FC<ProductCardProps> = ({ product, index }) => (
+//   <Link href={`/catalog/id=${index}`} passHref>
+//     <div key={product.id || index}>
+//       <div className="relative h-56">
+//         {product.masterVariant?.images?.[0]?.url ? (
+//           <Image
+//             layout="fill"
+//             objectFit="cover"
+//             src={product.masterVariant.images[0]?.url ?? ''}
+//             alt={product.masterVariant.name['en-US'] || 'Product image'}
+//           />
+//         ) : (
+//           <div className="flex items-center justify-center h-full bg-gray-200">
+//             <span>No Image Available</span>
+//           </div>
+//         )}
+//       </div>
+//       <div className="p-4">
+//         <h3 className="text-lg font-semibold text-center">
+//           {product.name['en-US']}
+//         </h3>
+//         {product.masterVariant.prices && (
+//           <p className="mt-2">
+//             Price: $
+//             {product.masterVariant.prices.filter(
+//               (price) => price.country === 'US',
+//             )[0]?.value.centAmount / 100}{' '}
+//             {
+//               product.masterVariant.prices.filter(
+//                 (price) => price.country === 'US',
+//               )[0]?.value.currencyCode
+//             }
+//           </p>
+//         )}
+//       </div>
+//     </div>
+//   </Link>
+// );
+
+// export default ProductCard;
+import React from 'react';
+import Image from 'next/image';
+import { ProductData } from './constants';
+import Link from 'next/link';
+
+interface ProductCardProps {
+  product: ProductData;
+  id: number;
+}
+
+const ProductCard: React.FC<ProductCardProps> = ({ product, id }) => (
```

Comment:
Such way of defining props type is very old-fashioned and  you will hardly find something like this today:

const ProductCard = ({ product, id }: ProductCardProps) - this is a better way ti write